plugins/processors/gpuattributes/processor.go

```diff
@@ -148,24 +82,48 @@ func (d *gpuAttributesProcessor) processMetrics(_ context.Context, md pmetric.Me
 				aggregatedMemoryMetric := d.awsNeuronMemoryMetricAggregator.FlushAggregatedMemoryMetric()
 				d.awsNeuronMetricModifier.ModifyMetric(aggregatedMemoryMetric, metrics)
 			}
+
+			//loop over all metrics and filter labels
+			for k := 0; k < metrics.Len(); k++ {
+				m := metrics.At(k)
+				d.processMetricAttributes(m)
+			}
 		}
+
+		dropResourceMetricAttributes(rs)
 	}
 	return md, nil
 }
 
-func (d *gpuAttributesProcessor) processGPUMetricAttributes(m pmetric.Metric) {
+func (d *gpuAttributesProcessor) processMetricAttributes(m pmetric.Metric) {
 	// only decorate GPU metrics
-	if !strings.Contains(m.Name(), gpuMetricIdentifier) {
+	isGpuMetric := strings.Contains(m.Name(), gpuMetricIdentifier)
+	isNeuronMetric := d.awsNeuronMetricModifier.IsProcessedNeuronMetric(m.Name())
+	if !isNeuronMetric && !isGpuMetric {
 		return
 	}
 
 	labelFilter := map[string]map[string]interface{}{}
-	if strings.HasPrefix(m.Name(), gpuContainerMetricPrefix) {
-		labelFilter = containerLabelFilter
-	} else if strings.HasPrefix(m.Name(), gpuPodMetricPrefix) {
-		labelFilter = podLabelFilter
-	} else if strings.HasPrefix(m.Name(), gpuNodeMetricPrefix) {
-		labelFilter = nodeLabelFilter
+	if isGpuMetric {
+		if strings.HasPrefix(m.Name(), containerMetricPrefix) {
+			labelFilter = metricFilters.ContainerLabelFilter
+		} else if strings.HasPrefix(m.Name(), podMetricPrefix) {
+			labelFilter = metricFilters.PodLabelFilter
+		} else if strings.HasPrefix(m.Name(), nodeMetricPrefix) {
+			labelFilter = metricFilters.NodeLabelFilter
+		}
+	} else if isNeuronMetric {
+		if strings.HasPrefix(m.Name(), containerMetricPrefix) {
+			labelFilter = metricFilters.ContainerNeuronMetricFilter
+		} else if strings.HasPrefix(m.Name(), podMetricPrefix) {
+			labelFilter = metricFilters.PodNeuronMetricFilter
+		} else if strings.HasPrefix(m.Name(), nodeMetricPrefix) {
+			labelFilter = metricFilters.NodeNeuronMetricFilter
+		}
+
+		if strings.Contains(m.Name(), "_neurondevice_hw") {
```

Comment:
Do we need to delete only for neuronDevice Hardware metrics? There are other node level metrics where we would need removing of labels too right?

Reply:
nope, the problem with the neurondevice metrics is that the observation count is low so the buffer for excess bytes is less, we can delete the labels from other node emf logs if we see any problem with them